numba/tests/test_array_constants.py

```diff
@@ -134,6 +134,32 @@ def pyfunc():
         out = cres.entry_point()
         self.assertEqual(out, 86)
 
+    def test_too_big_to_freeze(self):
+        """
+        Test issue https://github.com/numba/numba/issues/2188 where freezing
+        a constant array into the code thats prohibitively long and consume
+        too much RAM.
+        """
+        nelem = 10**7   # 10 million items
+        biggie = np.arange(nelem)
+
+        def pyfunc():
+            return biggie
```

Comment:
The fact that this is returning the original array (rather than a copy as is done for smaller arrays) may be surprising. Perhaps add a line in the docs about it?